File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    var answer = 0;
    console.log(clothes);
    var types = new Map();
    for(var [name,type] of clothes){
        if(types.has(type)){
            types.set(type,types.get(type)+1);
        }else{
            types.set(type,2);
        }
    }
    return [...types.values()].reduce((a,b)=>a*b)-1;
}

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String,Integer> cMap = new HashMap<String,Integer>();
        for(String[] s : clothes){
            String type = s[1];
            cMap.put(type,cMap.getOrDefault(type,0)+1);
        }
        for(String key : cMap.keySet()){
            answer*=cMap.get(key)+1;
        }
        return answer-1;
    }
}